RosNet/RosReader/RosBagReader.cs

```diff
@@ -47,7 +48,7 @@ public void Read(RosBag rosBag)
     {
         if (!File.Exists(rosBag.Path))
         {
-            throw new FileNotFoundException($"File with path {rosBag.Path} does not exist");
+            throw new FileNotFoundException($"File with path {rosBag.Path} does not exist", rosBag.Path);
         }
         using BinaryReader reader = new BinaryReader(File.Open(rosBag.Path, FileMode.Open));
```

Comment:
I think it is completely fair to let `File.Open`'s exception bubble up to the user, aka no reason to use `File.Exists` (due to race-conditions this can still throw an exception from `File.Open`, that is called time-of-check vs. time-of-use, which is a quite common source of bugs related to file-systems). If you wanted to wrap it in our own exception, you should rather use a `try-catch`

Reply:
And these exceptions should be documentated in the public API anyway for users to know what to expect